notebooks/working/ho_deconfounder_on_sim_data.py

```diff
@@ -0,0 +1,512 @@
+
+# mnl_specification["total_travel_distance"] = [1, 2, 3, 7, 8]
+# mnl_names["total_travel_distance"] = ['Travel Distance, units:mi (Drive Alone)',
+#                                       'Travel Distance, units:mi (SharedRide-2)',
+#                                       'Travel Distance, units:mi (SharedRide-3+)',
+#                                       'Travel Distance, units:mi (Walk)',
+#                                       'Travel Distance, units:mi (Bike)']
+
+# mnl_specification["cross_bay"] = [[2, 3], [4, 5, 6]]
+# mnl_names["cross_bay"] = ["Cross-Bay Tour (Shared Ride 2 & 3+)",
+#                           "Cross-Bay Tour (All Transit Modes)"]
+mnl_specification_noncausal["cross_bay"] = [[2, 3]]
+mnl_names_noncausal["cross_bay"] = ["Cross-Bay Tour (Shared Ride 2 & 3+)"]
+
+mnl_specification_noncausal["household_size"] = [[2, 3]]
+mnl_names_noncausal["household_size"] = ['Household Size (Shared Ride 2 & 3+)']
+
+mnl_specification_noncausal["num_kids"] = [[2, 3]]
+mnl_names_noncausal["num_kids"] = ["Number of Kids in Household (Shared Ride 2 & 3+)"]
+
+# +
+# Estimate the basic MNL model, using the hessian and newton-conjugate gradient
+mnl_model_noncausal = cm.create_choice_model(data=data,
+                                   alt_id_col="mode_id",
+                                   obs_id_col="observation_id",
+                                   choice_col="sim_choice",
+                                   specification=mnl_specification_noncausal,
+                                   model_type="MNL",
+                                   names=mnl_names_noncausal)
+
+num_vars_noncausal = len(reduce(lambda x, y: x + y, mnl_names_noncausal.values()))
+# Note newton-cg used to ensure convergence to a point where gradient 
+# is essentially zero for all dimensions. 
+mnl_model_noncausal.fit_mle(np.zeros(num_vars_noncausal),
+                  method="BFGS")
+
+# Look at the estimation results
+mnl_model_noncausal.get_statsmodels_summary()
+# -
+
+# ## Model 2
+
+# +
+mnl_specification_causal_1 = mnl_specification_noncausal.copy()
+mnl_names_causal_1 = mnl_names_noncausal.copy()
+
+mnl_specification_causal_1["confounder_all"] = [1, 2, 3]
+mnl_names_causal_1["confounder_all"] = ["Confounder - Drive alone",
+                                     "Confounder - Shared ride 2", 
+                                     "Confounder - Shared ride 3"]
+
+# +
+# Estimate the basic MNL model, using the hessian and newton-conjugate gradient
+mnl_model_causal_1 = cm.create_choice_model(data=data,
+                                   alt_id_col="mode_id",
+                                   obs_id_col="observation_id",
+                                   choice_col="sim_choice",
+                                   specification=mnl_specification_causal_1,
+                                   model_type="MNL",
+                                   names=mnl_names_causal_1)
```

Comment:
If there's no reason otherwise, I suggest following PEP8 [indentation guidelines](https://www.python.org/dev/peps/pep-0008/#indentation) by lining up all the arguments with `data=data`.